### PR TITLE
Docs: Add installing on LXD section

### PR DESCRIPTION
**Pull request by bschimke95**
Heavily inspired by: https://microk8s.io/docs/install-lxd

### CODE REVIEW

File: docs/src/howto/install/lxd.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+wget https://raw.githubusercontent.com/canonical/k8s-snap/main/tests/e2e/lxd-profile.yaml -O k8s.profile
+```
+
+We can now pipe that file into the LXD profile.
+
+```no-highlight
+cat k8s.profile | lxc profile edit k8s
+```
+
+And then clean up.
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:37]:
```suggestion
To remove the profile, execute:
```

File: docs/src/howto/install/lxd.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+Canonical Kubernetes requires some specific settings to work within LXD (these
+are explained in more detail below). These can be applied using a custom
+profile. The first step is to create a new profile:
+```no-highlight
+lxc profile create k8s
+```
+
+Once created, we’ll need to add the rules. 
+
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:26]: let's add a bit more here:
```suggestion
Get our pre-defined profile rules from github and save them as k8s.profile.
```

File: docs/src/howto/install/lxd.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+```
+
+## Start an LXD container for Canonical Kubernetes
+We can now create the container that Canonical Kubernetes will run in.
+
+```no-highlight
+lxc launch -p default -p k8s ubuntu:22.04 k8s
+```
+
+Note that this command uses the `default` profile, for any existing system
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:50]:
```suggestion
```{note} that this command uses the `default` profile, for any existing system
```

File: docs/src/howto/install/lxd.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+settings (networking, storage, etc.) before also applying the `k8s` profile -
+the order is important.
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:52]:
```suggestion
the order is important.
```
```

File: docs/src/howto/install/lxd.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+## Install Canonical Kubernetes in an LXD container
+First, we’ll need to install Canonical Kubernetes within the container.
+
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:56]: Would it be a good idea to link the install guide here?

File: docs/src/howto/install/lxd.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+```no-highlight
+lxc exec k8s -- sudo snap install k8s --classic
+```
+
+## Access Canonical Kubernetes services within LXD
+Assuming you left the default bridged networking when you initially setup LXD,
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:62]: Briefly add a bit of explanation/review for the user what default bridged networking does.

File: docs/src/howto/install/lxd.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+there is minimal effort required to access Canonical Kubernetes services inside
+the LXD container.
+
+Simply note the `eth0` interface IP address from
+
+```no-highlight
+lxc list k8s
+```
+
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:71]: I would add the output here and say in this example the IP is x

File: docs/src/howto/install/lxd.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
+# Install Canonical Kubernetes in LXD
+
+Canonical Kubernetes can also be installed inside an LXD container. This is a
+great way, for example, to test out clustered Canonical Kubernetes without the
+need for multiple physical hosts.
+
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:6]: Do we need prerequisits?

File: docs/src/howto/install/lxd.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+and use this to access services running inside the container.
+
+### Expose services to the container
+
+You’ll need to expose the deployment or service to the container itself before
+you can access it via the LXD container’s IP address. This can be done using
+`k8s kubectl expose`. This example will expose the deployment’s port 80 to a
+port assigned by Kubernetes.
+
+In this example, we will use [Microbot] as it provides a simple HTTP endpoint
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:81]: briefly explain what Microbot is

File: docs/src/howto/install/lxd.md [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+to expose. These steps can be applied to any other deployment.
+
+First, initialize the k8s cluster with 
+
+```no-highlight
+lxc exec k8s -- sudo k8s bootstrap
+```
+
+Now, let’s deploy Microbot (please note this image only works on `x86_64`).
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:90]:
```suggestion
Now, let's deploy Microbot (please be aware that this image is compatible only with x86_64 architecture).
```

File: docs/src/howto/install/lxd.md [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+service/kubernetes         ClusterIP   10.152.183.1     <none>        443/TCP        21m
+
+NAME                       READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/microbot   1/1     1            1           21m
+
+NAME                                  DESIRED   CURRENT   READY   AGE
+replicaset.apps/microbot-6d97548556   1         1         1       21m
+```
+
+As we can see, Microbot is running. Let’s expose it to the LXD container.
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:114]:
```suggestion
Now that Microbot is up and running, let's make it accessible to the LXD container by using the expose action.
```

File: docs/src/howto/install/lxd.md [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+
+## Explanation of custom LXD rules
+
+**boot.autostart: “true”**: Always start the container when LXD starts. This is
+needed to start the container when the host boots.
+
+**linux.kernel_modules**: Comma separated list of kernel modules to load before
+starting the container
+
+**lxc.apparmor.profile=unconfined**: Disable AppArmor. Allow the container to
```
Comment (louiseschmidtgen) [inline docs/src/howto/install/lxd.md:158]: After "Disable AppArmor" start with "AppArmor is responsible for..."

**Review by louiseschmidtgen — COMMENTED** [review]
Great work! 
I particularly enjoyed the attention to detail in the custom lxd rules section.
I've made some minor "beautify" suggestions

File: docs/src/howto/install/lxd.md (hunk 5 above)
Comment (bschimke95) [reply docs/src/howto/install/lxd.md:56]: Hm, there is not much to add to this installation command, is it? All necessary steps are explained below.

File: docs/src/howto/install/lxd.md (hunk 8 above)
Comment (bschimke95) [reply docs/src/howto/install/lxd.md:6]: I don't think so. The `multipass` install docs also do not have it. I'm not a big fan of putting something like "a good internet connection" as prerequisites.

File: docs/src/howto/install/lxd.md (hunk 9 above)
Comment (bschimke95) [reply docs/src/howto/install/lxd.md:81]: Let's let the link explain it :)

File: docs/src/howto/install/lxd.md [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+Containers. Containers need to be confined based on their profiles thus we rely
+on confining them and not the hosts. If you can account for the needs of the
+Containers you could tighten the AppArmor profile instead of disabling it
+completely, as suggested in [1].
+
+**lxc.cap.drop=**: Do not drop any capabilities [2]. For justification see
+above.
+
+**lxc.mount.auto=proc:rw sys:rw**: Mount proc and sys rw [3]. For privileged
+containers, lxc over-mounts part of /proc as read-only to avoid damage to the
```
[the remark this answers is not in the document]
Comment (bschimke95) [reply docs/src/howto/install/lxd.md:170]: mounting a mount over another

**Review by evilnick — APPROVED** [review]
:cake:

**Pull request merged**